app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/ProductPhotosFragment.java

```diff
@@ -81,7 +71,90 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
         final State state = (State) intent.getExtras().getSerializable("state");
         product = state.getProduct();
         mFragment = this;
+        // initialize the arraylist
+        imageNames = new ArrayList<>();
+        imagesRecycler = view.findViewById(R.id.images_recycler);
 
+
+        openFoodAPIClient.getImages(product.getCode(), new OpenFoodAPIClient.OnImagesCallback() {
+            @Override
+            public void onImageResponse(boolean value, String response) {
+
+                if (value && response != null) {
+
+                    // a json object referring to base json object
+                    JSONObject jsonObject = null;
+                    try {
+                        jsonObject = new JSONObject(response);
+                    } catch (JSONException e) {
+                        e.printStackTrace();
+                    }
+
+                    // a json object referring to images
+                    JSONObject images = null;
+                    try {
+                        images = jsonObject.getJSONObject("product").getJSONObject("images");
+                    } catch (JSONException e) {
+                        e.printStackTrace();
+                    }
+
+
+                    // loop through all the image names and store them in a array list
+                    for (int i = 0; i < images.names().length(); i++) {
+
+                        try {
+                            imageNames.add(images.names().getString(i));
+                        } catch (JSONException e) {
+                            e.printStackTrace();
+                        }
+
+
+                    }
+
+                    adapter = new ImagesAdapter(getContext(), imageNames, product.getCode(), ProductPhotosFragment.this::onImageClick);
+                    imagesRecycler.setAdapter(adapter);
+                    imagesRecycler.setLayoutManager(new GridLayoutManager(getContext(), 3));
+
+
+                }
+
+
+            }
+        });
+
+    }
+
+    public void openFullScreen(String mUrlImage) {
+        if (mUrlImage != null) {
+            Intent intent = new Intent(getContext(), FullScreenImage.class);
+            Bundle bundle = new Bundle();
+            bundle.putString("imageurl", mUrlImage);
+            intent.putExtras(bundle);
+            startActivity(intent);
+        }
+    }
+
+    @Override
+    public void onImageClick(int position) {
+        String baseUrlString = "https://static.openfoodfacts.org/images/products/";
+        String barcodePattern = new StringBuilder(product.getCode())
+                .insert(3, "/")
+                .insert(7, "/")
+                .insert(11, "/")
+                .toString();
+        String finalUrlString = baseUrlString + barcodePattern + "/" + imageNames.get(position) + ".400" + ".jpg";
```

Comment:
How will the barcode pattern work for EAN-8 barcodes? Inserting ` / ` at the 11th position doesn't make sense in that case.

Reply:
Yes. I did not know about it when i sent the PR. I dont know the splitting pattern for different types of barcodes. I have asked the question on github and no one has yet answered.

Reply:
cc @stephanegigandet